linkhash.c

```diff
@@ -685,5 +689,5 @@ int lh_table_delete(struct lh_table *t, const void *k)
 
 int lh_table_length(struct lh_table *t)
 {
-	return t->count;
+	return t == NULL ? -1 : t->count;
```

Comment:
Technically, this is a breaking change if this is a public method - but there are no comments in linkhask.h for this function.

I think the commit should add some (Doxygen) documentation in the header file to define this error condition.